src/transformStateWithClones.js

```diff
@@ -8,6 +8,31 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  const copyState = { ...state };
+  const result = [];
+
+  for (const action of actions) {
+    if (action.type === 'addProperties') {
+      Object.assign(copyState, action.extraData);
+    }
```

Comment:
[DONT REPEAT YOURSELF]: You are pushing the copy of the state to the result array in each condition. This should be done once after the switch statement to avoid repetition.